### ISSUE TITLE
Travis CI: Expected - Waiting for status to be reported

### ISSUE THREAD

**Issue by DimitriPapadopoulos** [adrienverge/openfortivpn#732]
@adrienverge I keep seeing this error message when creating a pull request. Can you check the repository settings and make sure it has been migrated as described below?

[Known Issue: Travis CI reports “Expected - Waiting for status to be reported” on the GitHub Status API, but the status never arrives](https://travis-ci.community/t/known-issue-travis-ci-reports-expected-waiting-for-status-to-be-reported-on-the-github-status-api-but-the-status-never-arrives/1154)

**Comment by DimitriPapadopoulos** [comment]
Perhaps have a look at the latest comments:
>[**BanzaiMan**](https://travis-ci.community/t/known-issue-travis-ci-reports-expected-waiting-for-status-to-be-reported-on-the-github-status-api-but-the-status-never-arrives/1154/22)
> Could you try syncing your data, https://travis-ci.com/account/repositories or https://travis-ci.org/account/repositories, and restart (portions of, if appropriate) the builds?

>[**BanzaiMan**](https://travis-ci.community/t/known-issue-travis-ci-reports-expected-waiting-for-status-to-be-reported-on-the-github-status-api-but-the-status-never-arrives/1154/27)
> **@Darth** If you have the authorization, could you try resetting the check status integration? (Disable and re-enable the repo?) You might need to restart the (parts of) the affected builds, unfortunately.

**Comment by DimitriPapadopoulos** [comment]
Alternatively can you integrate PR #731? Test `./tests/lint/run.sh` passes locally.

**Comment by adrienverge** [comment]
I tried what the latest comments suggest, but disabling the repo on https://travis-ci.org/account/repositories fails with: "*An error happened when we tried to alter settings on GitHub.*"

Anyway, I merged #731 and #733. Feel free to ping me if needed!

**Comment by DimitriPapadopoulos** [comment]
Perhaps you need to "revoke access in GitHub and log in again in Travis CI":
* [API Error when trying to travis enable a github repo](https://travis-ci.community/t/api-error-when-trying-to-travis-enable-a-github-repo/6628)
* [Pushes won’t trigger new builds anymore after the project has been inactive for a long time](https://travis-ci.community/t/pushes-wont-trigger-new-builds-anymore-after-the-project-has-been-inactive-for-a-long-time/5716/2)

**Comment by DimitriPapadopoulos** [comment]
Now #734 fails too. It looks like pull requests are currently out-of-order. If the above links don't work, we might have to open a ticket on the Travis site.  I cannot do that as I don't have access to the GitHub settings.

**Comment by adrienverge** [comment]
I just did that. Can you check again? I don't have a way to relaunch pending builds without closing then reopening pull requests.

**Comment by DimitriPapadopoulos** [comment]
Works just fine. Hopefully we won't get this Travis CI failure again.

**Comment by DimitriPapadopoulos** [comment]
Again all new PRs are "Waiting for status to be reported". I think this is relevant, but in theory you should have already applied relevant changes:
* [Known Issue: Travis CI reports “Expected - Waiting for status to be reported” on the GitHub Status API, but the status never arrives](https://travis-ci.community/t/known-issue-travis-ci-reports-expected-waiting-for-status-to-be-reported-on-the-github-status-api-but-the-status-never-arrives/1154/28)

**Comment by DimitriPapadopoulos** [comment]
See:
* #740
* #741

**Comment by DimitriPapadopoulos** [comment]
@adrienverge Travis CI still fails, see new pull request #755 for example. I cannot look into this (perhaps by opening an issue against Travis) because I cannot see or modify the set up of the repository.

**Comment by adrienverge** [comment]
@DimitriPapadopoulos I'm sorry about that :/

Last days, I experienced the same problem on other GitHub repositories. I'm hoping they will eventually fix it.

In the meantime, I removed the *merge protection*, which means you and Martin can merge even when tests didn't pass (or weren't reported).

**Comment by DimitriPapadopoulos** [comment]
I have found 11 Travis CI issues containing _Expected - Waiting for status to be reported_ and all of them are closed:
https://github.com/travis-ci/travis-ci/issues?q=is%3Aissue+Expected+-+Waiting+for+status+to+be+reported
Perhaps we should look into each of these issues and make sure we have applied all necessary changes.

Then we could open our own issue, but it looks like they rather expect such issues to be discussed on the [Travis CI Community](https://travis-ci.community/) forum these days.

In the meantime it might be worth trying `travis_retry`:
https://docs.travis-ci.com/user/common-build-problems/#timeouts-installing-dependencies

**Comment by DimitriPapadopoulos** [comment]
The most relevant post on the forum is probably this one:
[Known Issue: Travis CI reports “Expected - Waiting for status to be reported” on the GitHub Status API, but the status never arrives](https://travis-ci.community/t/known-issue-travis-ci-reports-expected-waiting-for-status-to-be-reported-on-the-github-status-api-but-the-status-never-arrives/1154)
Most other posts refer to this post.

**Comment by adrienverge** [comment]
TL;DR: It should work now :+1:

Thanks again @DimitriPapadopoulos, your last link is indeed the most relevant. Switching from travis-ci.**org** to travis-ci.**com** (hence from `continuous-integration/travis-ci` to `Travis CI - Pull Request` reports) should solve their bug, according to this post.

But their .org → .com transition is rather buggy, and it was impossible to "just enable" openfortivpn in travis-ci.com.  I encountered at least two of their bugs (for example most of my repos were listed, but some, including openfortivpn, didn't show up in the list, even while trying all the solutions from https://github.com/travis-ci/travis-ci/issues/8689, while revoking/re-granting all access from GitHub, and using private/clean browsers). I think this has something to do with the temporary `openfortivpn-ghsa-qcxx-7rxx-cgxx` repo, because this one randomly showed up/not showed up in the list.
Finally, I tried joining travis-ci.com beta, just in case. It allowed me to try an automatic migration of the openfortivpn repo.  Luckily, that finally worked... After 50 min spent on this problem.

The required check for pull requests in now the "new" one:
![Capture d’écran du 2020-07-22 15-25-45](https://user-images.githubusercontent.com/5244945/88181802-a612ca80-cc2f-11ea-9aaa-b9930b68decc.png)


PS: Since the old travis-ci.**org** integration is now revoked, old and current pull
requests don't have Travis reports (green icons) anymore.

PPS: I'll :crossed_fingers: and close this. Feel free to reopen if needed!

**Comment by DimitriPapadopoulos** [comment]
Excellent! Thank you for taking the time to look into this Travis CI mess.
